org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/generator/trace/TraceForStorageProvider.java

```diff
@@ -139,8 +140,13 @@ protected AbsoluteURI getAbsoluteLocation(IFile file) {
 	@Override
 	protected IProjectConfig getProjectConfig(IFile sourceFile) {
 		AbsoluteURI location = getAbsoluteLocation(sourceFile);
-		EclipseProjectConfigProvider projectConfigProvider = getServiceProvider(location).get(EclipseProjectConfigProvider.class);
```

Comment:
i wonder how this can ever happen. this method should be called for methods only that xtext knows by means of a iresourceserviceprovider. did you gain any clues about this?